### PR TITLE
sched_cb: Add function documentation

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds some documentation for the sched_cb module.

### Testing procedure

Read the docs :)

### Issues/PRs references

None

### CODE REVIEW

File: core/include/sched.h [hunk 1]
```diff
@@ -203,13 +203,25 @@ extern clist_node_t sched_runqueues[SCHED_PRIO_LEVELS];
  */
 NORETURN void sched_task_exit(void);
 
-#ifdef MODULE_SCHED_CB
+#if defined(MODULE_SCHED_CB) || defined(DOXYGEN)
```
Comment (miri64) [inline core/include/sched.h:206]:
```suggestion
#if IS_USED(MODULE_SCHED_CB) || defined(DOXYGEN)
```

?

Comment (bergzand) [reply core/include/sched.h]: Squashed this in.

File: core/include/sched.h [hunk 2]
```diff
@@ -203,13 +203,25 @@ extern clist_node_t sched_runqueues[SCHED_PRIO_LEVELS];
...
+ * @param   next        Pid of the next scheduled thread
+ */
+typedef void (*sched_callback_t)(kernel_pid_t active, kernel_pid_t next);
+
 /**
- *  @brief  Register a callback that will be called on every scheduler run
+ * @brief  Register a callback that will be called on every scheduler run
  *
- *  @param[in] callback The callback functions the will be called
+ * @param[in] callback The callback functions the will be called
```
Comment (fjmolinas) [inline core/include/sched.h:222]:
```suggestion
 * @param[in] callback The callback functions that will be called
```

Comment (bergzand) [reply core/include/sched.h]: Amended :)

**Pull request merged**